packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1238,17 +1349,19 @@ class RenderEditable extends RenderBox {
   void _paintContents(PaintingContext context, Offset offset) {
     assert(_textLayoutLastWidth == constraints.maxWidth);
     final Offset effectiveOffset = offset + _paintOffset;
-
-    if (_selection != null) {
+    if (_selection != null && !_floatingCursorOn) {
       if (_selection.isCollapsed && _showCursor.value && cursorColor != null) {
```

Comment:
Check the behavior on Android. This whole thing might be wrong altogether. In which case, let's just move this to after painting the text.

Reply:
What do you mean by testing this on Android? Android doesn't have any devices with pressure detecting screens so this conditional will always be true if selection isn't null.